app/src/main/java/com/mobiledev/news_app/domain/repository/NewsRepository.kt

```diff
@@ -2,9 +2,10 @@ package com.mobiledev.news_app.domain.repository
 
 import androidx.paging.PagingData
 import com.mobiledev.news_app.domain.model.Article
+import com.mobiledev.news_app.service.enums.Category
 import kotlinx.coroutines.flow.Flow
 
 interface NewsRepository {
 
-    fun getNewsPage(): Flow<PagingData<Article>>
+    fun getNewsPage(query: String? = null, category: Category?): Flow<PagingData<Article>>
```

Comment:
query should be builded here, from the top levels (view) should be passed users input.